contrib/babelfishpg_tsql/runtime/functions.c

```diff
@@ -1199,6 +1206,135 @@ object_id(PG_FUNCTION_ARGS)
 		PG_RETURN_NULL();
 }
 
+/*
+ * object_name
+ * 		returns the object name with object id and database id as input where database id is optional
+ * Returns NULL
+ * 		if there is no such object in specified database, if database id is not provided it will lookup in current database
+ * 		if user don't have right permission
+ */
+Datum
+object_name(PG_FUNCTION_ARGS)
+{
+	Oid 				object_id = PG_GETARG_OID(0);
+	Oid 				database_id = get_cur_db_id();
+	Oid 				user_id = GetUserId();
+	Oid				schema_id = InvalidOid;
+	HeapTuple 			tuple;
+	Relation			tgrel;
+	ScanKeyData 			key;
+	SysScanDesc 			tgscan;
+	EphemeralNamedRelation 		enr;
+	bool 				found = false;
+	char 				*result = NULL;
+
+	if(!PG_ARGISNULL(1))
+	{
+		char *db_name;
+		database_id = PG_GETARG_OID(1);
+		db_name = get_db_name(database_id);
+		if(db_name == NULL) /* database doesn't exist with given oid */
+			PG_RETURN_NULL();
+		user_id = GetSessionUserId();
```

Comment:
do we have test cases for both where user id is session user id vs user id?

Reply:
yes